src/lib/InstantSearch.ts

```diff
@@ -218,11 +213,16 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
       this._searchFunction = searchFunction;
     }
 
+    const defaultRoutingOptions = {
+      stateMapping: simpleStateMapping(),
+      router: historyRouter(),
+    };
```

Comment:
Is there a reason for moving this inside the function (declared here vs declared at the top) ? Is it depending on some kind of internal state?

Reply:
Originally  we wanted to use `singleIndexStateMapping` here, but it wasn't possible after all